admin/test/test_release.py

```diff
@@ -726,149 +713,25 @@ def update_repo(self, aws, yum,
-        """
-        Calling :func:`upload_rpms` with a documentation release version fails.
-        """
-        aws = FakeAWS(
-            routing_rules={},
-            s3_buckets={},
-        )
-        yum = FakeYum()
-
-        self.assertRaises(
-            DocumentationRelease,
-            self.upload_rpms, aws, yum,
-            self.rpm_directory, self.target_bucket, '0.3.0+doc1',
-            self.build_server)
-
-    def test_packages_uploaded(self):
-        """
-        Calling :func:`update_repo` uploads packages from a source repository
-        to S3.
-        """
-        aws = FakeAWS(
-            routing_rules={},
-            s3_buckets={
-                self.target_bucket: {},
-            },
-        )
-
-        self.update_repo(
-            aws=aws,
-            yum=FakeYum(),
-            rpm_directory=self.rpm_directory,
-            target_bucket=self.target_bucket,
-            target_key=self.target_key,
-            source_repo=self.create_fake_repository(files=self.repo_contents),
-            packages=self.packages,
-            flocker_version=self.dev_version,
-            distro_name=self.operating_systems[0]['distro'],
-            distro_version=self.operating_systems[0]['version'],
-        )
-
-        self.assertDictContainsSubset(
-            {os.path.join(self.target_key, package):
-                self.repo_contents[package]
-             for package in self.repo_contents},
-            aws.s3_buckets[self.target_bucket])
-
-    def test_metadata_uploaded(self):
-        """
-        Calling :func:`update_repo` uploads metadata for a source repository to
-        S3.
+    def test_fake_rpm(self):
         """
-        aws = FakeAWS(
-            routing_rules={},
-            s3_buckets={
-                self.target_bucket: {},
-            },
-        )
+        Calling :func:`update_repo` downloads the new RPMs, creates the
+        metadata, and uploads it to S3.
 
-        self.update_repo(
-            aws=aws,
-            yum=FakeYum(),
-            rpm_directory=self.rpm_directory,
-            target_bucket=self.target_bucket,
-            target_key=self.target_key,
-            source_repo=self.create_fake_repository(files=self.repo_contents),
-            packages=self.packages,
-            flocker_version=self.dev_version,
-            distro_name=self.operating_systems[0]['distro'],
-            distro_version=self.operating_systems[0]['version'],
-        )
-
-        self.assertIn(
-            os.path.join(self.target_key, 'repodata', 'repomd.xml'),
-            aws.s3_buckets[self.target_bucket])
-
-    def test_repository_added_to(self):
-        """
-        Calling :func:`update_repo` does not delete packages or metadata which
-        already exist in S3.
+        - Existing package on S3 are preserved in the metadata.
+        - Other packages on the buildserver are not downloaded.
+        - Existing metadata files are left untouched.
```

Comment:
There is a trade-off here. We know exactly what we want the final state of the bucket to be. We can either assert that is what we want, or we could write a bunch of assertions that assert the same thing. In the former, it is slightly less clear what specific things are being tested for each part. On the other hand, the latter is more complicated to write and (I think) maintain.